packages/pn-personafisica-webapp/src/components/Contacts/PecContactItem.tsx

```diff
@@ -58,18 +85,53 @@ const PecContactItem = ({ value, verifyingAddress, blockDelete }: Props) => {
     await formik.setFieldTouched(e.target.id, true, false);
   };
 
-  const handleEditConfirm = (status: 'validated' | 'cancelled') => {
-    if (status === 'cancelled') {
-      formik.resetForm({ values: initialValues });
+  const handleCodeVerification = (verificationCode?: string) => {
+    if (verificationCode) {
+      PFEventStrategyFactory.triggerEvent(PFEventsType.SEND_ADD_PEC_UX_CONVERSION, 'default');
     }
-  };
 
-  const handlePecValidationCancel = () => {
-    setCancelDialogOpen(true);
+    const digitalAddressParams: SaveDigitalAddressParams = {
+      addressType: AddressType.LEGAL,
+      senderId: 'default',
+      channelType: ChannelType.PEC,
+      value: formik.values.pec,
+      code: verificationCode,
+    };
+
+    void dispatch(createOrUpdateAddress(digitalAddressParams))
+      .unwrap()
+      .then((res) => {
+        // contact to verify
+        // open code modal
+        if (!res) {
+          setModalOpen(ModalType.CODE);
+          return;
+        }
+
+        PFEventStrategyFactory.triggerEvent(PFEventsType.SEND_ADD_PEC_UX_SUCCESS, 'default');
+
+        // contact has already been verified
+        if (res.pecValid) {
+          // show success message
+          dispatch(
+            appStateActions.addSuccess({
+              title: '',
+              message: t(`legal-contacts.pec-added-successfully`, { ns: 'recapiti' }),
+            })
+          );
+          setModalOpen(null);
+          digitalElemRef.current.toggleEdit();
```

Comment:
When I add a new PEC, the field remains in edit mode even if the operation is successful.
However, when I edit an existing PEC, the behaviour is correct.